adapters/repos/db/vector/cache/sharded_lock_cache.go

```diff
@@ -19,21 +19,20 @@ import (
 	"unsafe"
 
 	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/common"
 	"github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/distancer"
 )
 
-type shardedLockCache struct {
-	shardedLocks        []sync.RWMutex
-	cache               [][]float32
-	vectorForID         VectorForID
-	normalizeOnRead     bool
-	maxSize             int64
-	count               int64
-	cancel              chan bool
-	logger              logrus.FieldLogger
-	dims                int32
-	trackDimensionsOnce sync.Once
-	deletionInterval    time.Duration
+type shardedLockCache[T float32 | byte | uint64] struct {
+	shardedLocks     []sync.RWMutex
+	cache            [][]T
```

Comment:
I know this is existing implementation, but I'm not sure I fully get how it works. If single vector of id 1_000_000 needs to be cached it requires underlying slice of len > 1_000_000?

Reply:
Correct. It assumes that you mostly have contiguous IDs (which would be the case after a bulk import with zero deletes). Could probably optimized in case there are larger gaps.